cmake/app.cmake

```diff
@@ -1,23 +1,26 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+# Create Mbed OS library
+add_library(mbed-os OBJECT)
```

Comment:
I'm not sure about this. We should be adding the targets in the `CMakeLists.txt` entry points. A script named `app.cmake` shouldn't be responsible for adding the `mbed-os` library target. I believe we should also be adding the `app` executable in the application `CMakeLists.txt`. This is the standard CMake pattern and I think we should try to stick to it.

Reply:
this is because of adding linking options to a target down in Mbed OS , @hugueskamba is that correct? 

As stated above, it might be better to just rework the log to provide series of commits with  reasoning to answer our questions.

Reply:
You cannot add options (link or compile) to a target that has not yet been created hence why the executable and library creation statements were moved before the include statements (which call the `target_` functions down the line).

Reply:
The latest commit makes this redundant ?